rust/src/applayernewmodbus/newmodbus.rs

```diff
@@ -570,9 +736,30 @@ pub unsafe extern "C" fn rs_newmodbus_register_parser() {
 #[cfg(test)]
 mod test {
     use super::*;
+    use crate::applayer::AppLayerResult;
+    use crate::core;
+    use std::cmp;
+    use std::mem;
+
+    #[test]
+    fn test_header_constructor() {
+        let mut header_valid = NewModbusHeader::new();
+        header_valid.transaction_id = 0x000A;
+        header_valid.protocol_id = MODBUS_PROTOCOL_VER;
+        header_valid.length = 0x0008;
+        header_valid.unit_id = 0x00;
+
+        assert_eq!(7,mem::size_of::<NewModbusHeader>());
+    }
+
+    #[test]
+    fn test_max_pdu_length() {
+        // TODO what's the best way to test this?
+        unimplemented!("Still to understand best way to check this. PCAPs?");
```

Comment:
I think you can also do this with unittests. But, you'll have to have a header parsing function first that parses the length of a PDU from the protocol header.

Reply:
Thanks, makes sense!